src/collection.ts

```diff
@@ -1072,14 +1072,18 @@ export class Collection<TSchema extends Document = Document> {
   ): ListSearchIndexesCursor {
     options =
       typeof indexNameOrOptions === 'object' ? indexNameOrOptions : options == null ? {} : options;
+
+    /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+    const cleanedOptions = (({ readConcern, writeConcern, ...rest }) => rest)(options);
```

Comment:
whoa what a clever way to remove fields from an object!  I've never seen this before but I like it 🙂

Reply:
```suggestion
    const { readConcern, writeConcern, ...cleanedOptions } = options;
```

What is the purpose of the closure?

Reply:
No reason for this extra layer, removed it.

Edit: Removed the entire line of code since with the new type checking in the argument it is no longer relevant.

Reply:
I'm not sure I fully agree with typescript being the final source of trust that we don't attach those settings but willing to be convinced otherwise. I feel as though the one liner to prevent the error is worth keeping. @baileympearson thoughts?

Reply:
When I add the line back in it says that the options type does not contain readConcern / writeConcern with the updated options typing.

Reply:
I don't feel strongly.  We do, however, have precedent for removing write concern from options where it should never exist according to TS: https://github.com/mongodb/node-mongodb-native/blob/918fe69d104d8573966d52605b916a9de904d11a/src/operations/indexes.ts#L358